src/cython/gudhi/rips_complex.pyx

```diff
@@ -93,6 +96,7 @@ cdef class RipsComplex:
         :returns: A simplex tree created from the Delaunay Triangulation.
         :rtype: SimplexTree
         """
-        simplex_tree = SimplexTree()
-        self.thisref.create_simplex_tree(simplex_tree.thisptr, max_dimension)
-        return simplex_tree
+        stree = SimplexTree()
+        cdef intptr_t stree_ptr=stree.thisptr
+        self.thisref.create_simplex_tree(<Simplex_tree_interface_full_featured*>stree_ptr, max_dimension)
```

Comment:
Ah, you did use one line fewer here :wink:

Reply:
You are right. Done here: 3c96720c786462827820f4d35e720c8f2084867b